### PR TITLE
Add simple API for the Room engine

### PR DESCRIPTION
**Pull request by fbarthelery**

### CODE REVIEW

**Comment by genypaul** [comment]
LGTM

File: mobile/src/main/java/com/genymobile/roomconnect/roomengine/ConnectionListener.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package com.genymobile.roomconnect.roomengine;
+
+
+public interface ConnectionListener {
```
Comment (flo-genymobile) [inline mobile/src/main/java/com/genymobile/roomconnect/roomengine/ConnectionListener.java]: what is connection? 
would be authentication moar clear?

File: mobile/src/main/java/com/genymobile/roomconnect/roomengine/JoinRoomListener.java [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package com.genymobile.roomconnect.roomengine;
+
+
+public interface JoinRoomListener {
+
+    void onRoomJoinSuccessfull();
+
+    void onRoomJoinError();
```
Comment (flo-genymobile) [inline mobile/src/main/java/com/genymobile/roomconnect/roomengine/JoinRoomListener.java]: the interface name is prefixed with `JoinRoom` but the method are prefixed with `RoomJoin`. Choose :-)

File: mobile/src/main/java/com/genymobile/roomconnect/roomengine/ConnectionListener.java (hunk 1 above)
Comment (fbarthelery) [inline mobile/src/main/java/com/genymobile/roomconnect/roomengine/ConnectionListener.java]: The goal is for the ServiceModule to be connected and able to interact with its service (Slack for example). Some service may not required authentication. Even for slack there is no authentication, just a hardcoded token to use.

File: mobile/src/main/java/com/genymobile/roomconnect/roomengine/JoinRoomListener.java (hunk 2 above)
Comment (fbarthelery) [inline mobile/src/main/java/com/genymobile/roomconnect/roomengine/JoinRoomListener.java]: Renamed :+1:

File: mobile/src/main/java/com/genymobile/roomconnect/roomengine/ConnectionListener.java (hunk 1 above)
Comment (flo-genymobile) [inline mobile/src/main/java/com/genymobile/roomconnect/roomengine/ConnectionListener.java]: :+1:

**Comment by flo-genymobile** [comment]
:sheep: :it:

**Comment by flo-genymobile** [comment]
@fbarthelery @genypaul i dont have right to merge :(

**Comment by fbarthelery** [comment]
@flo-genymobile You should have it now

**Pull request merged**